ci/CONTRIBUTING.MD

```diff
@@ -0,0 +1,39 @@
+- [AWS CloudFormation Template Configuration File Reference](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/continuous-delivery-codepipeline-cfn-artifacts.html#d0e10050)
+- [AWS CodeBuild Build Specification Reference](https://docs.aws.amazon.com/codebuild/latest/userguide/build-spec-ref.html)
+- [AWS CodeBuild Build Phase Transitions](https://docs.aws.amazon.com/codebuild/latest/userguide/view-build-details.html#view-build-details-phases)
+- [AWS CodeBuild Build Environment Reference](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref.html)
+- [AWS CodeBuild Environment Variables in Build Environments ](https://docs.aws.amazon.com/codebuild/latest/userguide/build-env-ref-env-vars.html)
+
+## Deployment
+
+The continuous integration platform is intended to be managed and launched through [AWS CloudFormation](https://aws.amazon.com/cloudformation/), so any AWS resources that are necessary for the system to operate should be added to the `ci.yml` template that already exists. CloudFormation stacks should always be designed to work in any AWS region, and allow for multiple stacks to be launched without conflicting, even in the same region (though doing so is unexpected and not recommended).
+
+**Never make changes changes to AWS resources that were created through CloudFormation, and do not create any resources outside of the CloudFormation template**. The CI system is entirely self-contained and should remain that way.
+
+Sensitive information, such as API keys, should be managed through [Parameter Store](http://docs.aws.amazon.com/systems-manager/latest/userguide/systems-manager-paramstore.html) when possible. In cases where that's not possible, the values can be passed in to the CloudFormation stack as normal [stack parameters](http://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/parameters-section-structure.html).
+
+When there are external dependencies, such as the configuration of GitHub Webhooks, the [README](https://github.com/PRX/Infrastructure/blob/master/ci/README.md) should include information to help with getting those dependencies set up.
+
+There is no automated process for deploying changes to the CI stack. If the CloudFormation template changes, the stack must be updated manually (through the AWS Console or command line). You should always be sure to carefully review the resources that are changing before executing the stack update. The code for AWS Lambda functions likewise will not deploy automatically, even if you deploy stack changes manually. Even if you push updated Lambda code to S3 prior to executing a stack update, CloudFormation may not always notice the changes. For now, the easiest and most reliable way to deploy Lambda code changes is copy-and-paste through the console. (This is not true of other properties of Lambda functions defined in the template, such as the timeout; those will update as expected during a stack update.)
```

Comment:
any guidance for what steps to take as part of how to "carefully review the resources that are changing before executing the stack update" ? or any must-do steps?

Reply:
also would be good to add a little more clarity around Lambda deploy -- easiest + best way currently is to copy + paste through console; do we _also_ need to push updated code to s3? and/or update the s3 code object version ID in secrets anywhere?